Sources/AvatarView.swift

```diff
@@ -38,17 +47,19 @@ open class AvatarView: UIView {
     }
     
     convenience public init() {
-        let frame = CGRect(x: 0, y: 0, width: 30, height: 30)
-        self.init(frame: frame)
-        prepareView()
+        self.init(frame: .zero)
     }
     
     private func getImageFrom(initals: String, withColor color: UIColor = UIColor.white, fontSize: CGFloat = 14) -> UIImage {
-        _ = UIGraphicsBeginImageContext(CGSize(width: 30, height: 30))
+        let width = frame.width
+        let height = frame.height
+        if width == 0 || height == 0 {return #imageLiteral(resourceName: "Steve-Jobs")}
```

Comment:
We definitely shouldn't be returning a default value of `Steve-Jobs` here :joy:
We should just do away with checking the image height/width entirely

Reply:
God :) I though I have fixed that issue :) I am sorry about that. One version I fixed it but I think I copied from the wrong one :/

Reply:
I was trying to return UIImage()

Reply:
I did that check, because I did not want to do extra work below if we have CGZero. The code initialized with CGZero and then update the frame based on the delegate method.

Reply:
If it's a debug build could we do something to print an error (or maybe even fatal in debug to make it clear) because the avatar size should never be zero, right?

Reply:
@Sherlouk We actually support an avatar size of zero if the user doesn't want an avatar in their cell. I'm open to another way of doing it but I thought it would be cheaper to hide the existing view rather than initialize `UIView`s and add them to the cell in `cellForItem(at:indexPath:)`

Reply:
You guys are bringing up some great points. @gungorbasa We should keep this check so we don't perform the drawing code if the size is zero.

Reply:
@SD10 1 Question. For the rounded corners, should I consider the case of giving corner_radius > width/2. In this case, avatar picture can get really weird. 

<img width="505" alt="screen shot 2017-09-11 at 6 27 01 am" src="https://user-images.githubusercontent.com/2487031/30276672-6a1e8d14-96ba-11e7-9e10-14df5326dc26.png">
<img width="487" alt="screen shot 2017-09-11 at 6 27 52 am" src="https://user-images.githubusercontent.com/2487031/30276671-6a16f75c-96ba-11e7-82d6-79ecc4bf3295.png">

Reply:
@gungorbasa I don't think we should worry about anything more than rounded rects or circular avatars. I've never run into a situation where I've used anything other than this